thunder/tests/test_fa3_executor.py

```diff
@@ -0,0 +1,237 @@
+    k = torch.randn([batch, seq_len, num_heads, dim_per_head], device="cuda", dtype=dtype, requires_grad=True)
+    v = torch.randn([batch, seq_len, num_heads, dim_per_head], device="cuda", dtype=dtype, requires_grad=True)
+
+    def fn(query, key, value):
+        return torch.nn.functional.scaled_dot_product_attention(query, key, value)
+
+    cfn = thunder.jit(fn, executors=[fa3_ex])
+
+    # Verifies the result is close to PyTorch
+    out = cfn(q, k, v)
+    out_ref = attention_ref(q, k, v)[0]
+    out_pt = attention_ref(
+        q,
+        k,
+        v,
+        upcast=False,
+        reorder_ops=True,
+    )[0]
+
+    g = torch.randn_like(out)
+    do_o = (g.float() * out.float()).sum(-1)
+    dq, dk, dv = torch.autograd.grad(out, (q, k, v), g)
+    dq_ref, dk_ref, dv_ref = torch.autograd.grad(out_ref, (q, k, v), g)
+    dq_pt, dk_pt, dv_pt = torch.autograd.grad(out_pt, (q, k, v), g)
+    assert (dq - dq_ref).abs().max().item() <= 2 * (dq_pt - dq_ref).abs().max().item() + 3e-5
+    assert (dk - dk_ref).abs().max().item() <= 2 * (dk_pt - dk_ref).abs().max().item() + 3e-5
+    assert (dv - dv_ref).abs().max().item() <= 2 * (dv_pt - dv_ref).abs().max().item() + 3e-5
+
+
+# verify that fa3 kernel is properly being called when used in a valid trace
+@requiresCUDA
+def test_fa3_used():
+    if not HAS_FA3:
+        pytest.skip("fa3 not built")
+
+    batch = 4
+    seq_len = 128
+    num_heads = 6
+    dim_per_head = 64
+    device = "cuda"
+    dtype = torch.float16
+
+    query = torch.randn([batch, seq_len, num_heads, dim_per_head], device="cuda", dtype=dtype, requires_grad=True)
+    key = torch.randn([batch, seq_len, num_heads, dim_per_head], device="cuda", dtype=dtype, requires_grad=True)
+    value = torch.randn([batch, seq_len, num_heads, dim_per_head], device="cuda", dtype=dtype, requires_grad=True)
+
+    def fn(query, key, value):
+        return torch.nn.functional.scaled_dot_product_attention(query, key, value)
+
+    cfn = thunder.jit(fn, executors=[fa3_ex])
+
+    thunder_result = cfn(query, key, value)
+
+    # Verifies fa3 fwd was called
+    extrace = thunder.last_traces(cfn)[-1]
+    assert any(bsym.sym.name == "fa3_fwd" for bsym in extrace.bound_symbols)
+
+    """
+    loss = thunder_result.sum()
+    loss.backward()
+
+    # Verifies fa3 bwd was called
+    extrace = thunder.last_traces(cfn)[-1]
+    print(f'symbols: {[bsym.sym.name for bsym in extrace.bound_symbols]}')
+    assert any(bsym.sym.name == "fa3_bwd" for bsym in extrace.bound_symbols)
+    """
+
+
+# verify that checker is correctly returning False on invalid fa3 use cases
+@requiresCUDA
+def test_checker():
+
+    if not HAS_FA3:
+        pytest.skip("fa3 not built")
+
+    batch = 10
+    seq_len = 128
+    num_heads = 4
+    dim_per_head = 32
+
+    device = "cuda"
+    dtype = torch.float16
+    attn_mask = None
+    dropout_p = 0.0
+
+    def check(device, dtype, attn_mask, dropout_p):
```

Comment:
nit: a comment describing this function would be helpful

Reply:
Added